language-server/src/dotty/tools/languageserver/DottyLanguageServer.scala

```diff
@@ -343,15 +353,22 @@ class DottyLanguageServer extends LanguageServer
 
     val pos = sourcePosition(driver, uri, params.getPosition)
     val uriTrees = driver.openedTrees(uri)
-    val sym = Interactive.enclosingSourceSymbol(uriTrees, pos)
+    val enclTree = Interactive.enclosingTree(driver.openedTrees(uri), pos)
 
-    if (sym == NoSymbol) Nil.asJava
-    else {
-      val refs = Interactive.namedTrees(uriTrees, Include.references | Include.overriding, sym)
-      ( for (ref <- refs if ref.namePos.exists)
-        yield new DocumentHighlight(range(ref.namePos), DocumentHighlightKind.Read)
-      ).asJava
+    val syms =
+      enclTree match {
```

Comment:
Having various places where we special case what to do based on whether `enclTree` is an import or not seems fragile. Can we instead generalize `Interactive.enclosingSourceSymbol` to work with imports (and return a list of symbols instead of a single symbol) ?